src/npc.h

```diff
@@ -221,6 +227,7 @@ class Npc final : public Creature
 		std::map<std::string, std::string> m_parameters;
 
 		std::set<Player*> shopPlayerSet;
+		std::set<Player*> spectators;
```

Comment:
What is the reason for keeping a set of spectators? You might as well use a simple counter.